### DISCUSSION TITLE
Is there any way to scroll to a specific item to make it visible when the chart is rendered?

### DISCUSSION THREAD

**Discussion by alexandre-thauvin** [patrykandpatrick/vico#265] [Ideas]
Hello! I have a set of entries (in a simple `columnChart`) I have certain value Y value and I want to scroll to this specific Y value. Is there any way to achieve that?

I saw all the code about perform `ChartScrollSpec` (including `performAutoScroll`), `ChartScrollState` and the only functions to perform a programmatic scroll is `scrollby/animateScrollBy` and the value in parameter is to scroll via a number of pixels which is quite not convenient... 
If any example can be provided that would be awesome!

**Comment by patrickmichalik** [comment]
Hello! We’ll be implementing a feature that’ll make this possible. The plan is to introduce functions for scrolling (1) by a specific number of entries and (2) to the entry corresponding to a given _x_ value. The second function will take an _x_ value instead of a _y_ value because _y_ values aren’t necessarily unique. However, for injective mappings, if a given value appears in the codomain, one can easily find the corresponding value from the domain. I’ll move this discussion to the “Ideas” category.

**Reply by alexandre-thauvin** [reply]
Hi! That’s awesome to hear!!
Your work is greatly appreciated! Thank you guys 🔥🙏

**Reply by MarioMitrevski** [reply]
Great to hear this! Can you tell us when we can expect this feature? Thank you for providing us this great library.

**Reply by patrickmichalik** [reply]
Thank you for the kind words. Because Vico is a free project that we work on in our spare time, I’m afraid that we’re unable to provide a timeline, @MarioMitrevski. However, this is a relatively small change, so I suspect we’ll introduce it sooner than later.

**Comment by patrickmichalik** [comment]
[Vico 2.0.0 Alpha 9](https://github.com/patrykandpatrick/vico/releases/tag/v2.0.0-alpha.9) introduces this functionality. `Scroll.Absolute.x` scrolls to a specific _x_ coordinate, and the `bias` parameter lets you set the coordinate’s alignment. For example, the following ensures that _x_ = 10 is initially visible and centered. Cheers!

```kt
rememberVicoScrollState(initialScroll = Scroll.Absolute.x(x = 10f, bias = .5f))
```

**Comment by Gowsky** [comment]
Hello, [Vico 2.0.0](https://github.com/patrykandpatrick/vico/releases/tag/v2.0.0) is now available.
